worlds/tloz/Client.py

```diff
@@ -0,0 +1,376 @@
+            current_container_count = min(((current_heart_byte_value & 0xF0) >> 4) + 1, 15)
+            current_hearts_count = min((current_heart_byte_value & 0x0F) + 1, 15)
+            self.guard_list.append((Rom.heart_containers, [current_heart_byte_value], self.wram))
+            write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_hearts_count], self.wram))
+        elif item_name == "Triforce Fragment":
+            current_triforce_value = await self.read_ram_value_guarded(ctx, Rom.triforce_count)
+            if current_triforce_value is None:
+                return
+            self.guard_list.append((Rom.triforce_count, [current_triforce_value], self.wram))
+            write_list.append((Rom.triforce_count, [min(current_triforce_value + 1, 8)], self.wram))
+        elif item_name == "Boomerang":
+            write_list.append((Rom.boomerang, [1], self.wram))
+        elif item_name == "Magical Boomerang":
+            write_list.append((Rom.magical_boomerang, [1], self.wram))
+        elif item_name == "Magical Shield":
+            write_list.append((Rom.magical_shield, [1], self.wram))
+        elif item_name == "Recovery Heart":
+            current_heart_byte_value = await self.read_ram_value_guarded(ctx, Rom.heart_containers)
+            if current_heart_byte_value is None:
+                return
+            self.guard_list.append((Rom.heart_containers, [current_heart_byte_value], self.wram))
+            current_container_count = ((current_heart_byte_value & 0xF0) >> 4)
+            current_hearts_count = current_heart_byte_value & 0x0F
+            if current_hearts_count >= current_container_count:
+                write_list.append((Rom.partial_hearts, [0xFF], self.wram))
+                write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_container_count], self.wram))
+            else:
+                current_hearts_count = min(current_hearts_count + 1, 15)
+                write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_hearts_count], self.wram))
+        elif item_name == "Fairy":
+            current_heart_byte_value = await self.read_ram_value_guarded(ctx, Rom.heart_containers)
+            if current_heart_byte_value is None:
+                return
+            self.guard_list.append((Rom.heart_containers, [current_heart_byte_value], self.wram))
+            current_container_count = ((current_heart_byte_value & 0xF0) >> 4)
+            current_hearts_count = current_heart_byte_value & 0x0F
+            if (current_hearts_count + 3) >= current_container_count:
+                write_list.append((Rom.partial_hearts, [0xFF], self.wram))
+                write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_container_count], self.wram))
+            else:
+                write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_hearts_count], self.wram))
```

Comment:
Fairy item currently doesn't actually heal.
```suggestion
            else:
                current_hearts_count = min(current_hearts_count + 3, 15)
                write_list.append((Rom.heart_containers, [(current_container_count << 4) | current_hearts_count], self.wram))
```

Reply:
oops